### PR TITLE
Add avalanche and solana tokens to asset type

### PR DESCRIPTION
**Pull request by yabirgb**

### CODE REVIEW

**Comment by yabirgb** [comment]
The failing test `rotkehlchen/tests/integration/test_blockchain.py::test_multiple_concurrent_ethereum_blockchain_queries` works locally so must be a timeout in the CI

**Review by LefterisJP — COMMENTED** [review]
> The failing test rotkehlchen/tests/integration/test_blockchain.py::test_multiple_concurrent_ethereum_blockchain_queries works locally so must be a timeout in the CI

You can't just assume that. You have to look at the logs. From which you see:

```
>       assert blockchain.totals.assets[A_DAI].amount == 2
E       assert FVal(10000000001) == 2
E        +  where FVal(10000000001) = Balance(amount=FVal(10000000001), usd_value=FVal(15000000001.0)).amount
```

which is very much NOT a timeout.

Lucky for you, it's the test I fixed in https://github.com/rotki/rotki/pull/2727. You are not rebased on latest develop.

I have not added a new type in a long time so I am not sure anymore if this is the only place. But you are definitely missing the DB type addition in the global db schema.

https://github.com/rotki/rotki/blob/2bb2f35fb13acee2f5994f3f2c3dc74c5f89a2a7/rotkehlchen/globaldb/schema.py#L21-L73

I thought we had a test for this. It seems we do not. Can you add a unit test in test_globaldb.py that check the `AssetType` enum has corresponding entries in the DB?

**Comment by yabirgb** [comment]
@LefterisJP Okey, since I was getting this value that I remember seeing in the mocks somewhere I thought it was using a default value instead of the fetched one. 

For the DB I didn't find the table and wanted the CI to alert me. Will also add a test for this

**Comment by yabirgb** [comment]
@LefterisJP i've rebased and added the locations to the `DB_CREATE_ASSET_TYPES` sql sentence but seems that is not being executed. Is there any additional step needed to update the globaldb?

**Review by LefterisJP — COMMENTED** [review]
> @LefterisJP i've rebased and added the locations to the DB_CREATE_ASSET_TYPES sql sentence but seems that is not being executed. Is there any additional step needed to update the globaldb?

In the tests we have this fixture:  https://github.com/rotki/rotki/blob/5c01fffa4933e3c9c4586eb6b2665d19469f88cb/rotkehlchen/tests/fixtures/globaldb.py#L30-L43

It copies the latest premade DB to the test directory so it can be used. But it should work. Since it should also run the initialization as seen in here: https://github.com/rotki/rotki/blob/5c01fffa4933e3c9c4586eb6b2665d19469f88cb/rotkehlchen/globaldb/handler.py#L51-L63

And in the initialization you added the two new types. I would have to clone and debug myself to figure out why.

File: rotkehlchen/globaldb/schema.py [hunk 1]
```diff
@@ -70,6 +70,10 @@
 INSERT OR IGNORE INTO asset_types(type, seq) VALUES ('V', 22);
 /* OTHER */
 INSERT OR IGNORE INTO asset_types(type, seq) VALUES ('W', 23);
+/* AVALANCHE TOKEN */
+INSERT OR IGNORE INTO asset_types(type, seq) VALUES ('X', 24);
+/* SOLANA TOKEN */
+INSERT OR IGNORE INTO asset_types(type, seq) VALUES ('Y', 23);
```
Comment (LefterisJP) [inline rotkehlchen/globaldb/schema.py:76]: Hmm I think your problem is here :sweat: 

```suggestion
INSERT OR IGNORE INTO asset_types(type, seq) VALUES ('Y', 25);
```

File: rotkehlchen/tests/db/test_db.py [hunk 2]
```diff
@@ -1352,3 +1353,17 @@ def test_values_are_present_in_db(database, enum_class, table_name):
...
+
+
+@pytest.mark.parametrize('enum_class, table_name', [(AssetType, 'asset_types')])
+def test_values_are_present_in_global_db(globaldb, enum_class, table_name):
+    """
+    Check that all enum classes have the same number of possible values
+    in the class definition as in the database
+    """
+    cursor = globaldb._conn.cursor()
+    query = 'SELECT COUNT(*) FROM {} WHERE seq=?'.format(table_name)
```
Comment (LefterisJP) [inline rotkehlchen/tests/db/test_db.py:1365]: unless it's a log entry (where for optimization we should switch to using the lazily initialized string) please use f-strings

File: rotkehlchen/tests/db/test_db.py [hunk 3]
```diff
@@ -1352,3 +1353,17 @@ def test_values_are_present_in_db(database, enum_class, table_name):
...
+
+    for enum_class_entry in enum_class:
+        r = cursor.execute(query, (enum_class_entry.value,))
+        assert r.fetchone() == (1,)
```
Comment (LefterisJP) [inline rotkehlchen/tests/db/test_db.py:1369]: Here you could have found the problem faster if you had an assertion error message

```suggestion
        assert r.fetchone() == (1,), f'Did not find {table_name} entry for value {enum_class_entry.value}'
```

**Review by LefterisJP — COMMENTED** [review]
Run the tests locally before pushing.

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**